src/profile/Signup.js

```diff
@@ -12,93 +15,86 @@ import LoaderButton from "../components/LoaderButton";
  * This component signs you up for the Cognito user
  * @TODO Issue #7
  * 
- * import useState
- *    isLoading: false,
-      email: "",
-      password: "",
-      confirmPassword: "",
-      confirmationCode: "",
-      newUser: null,
+ * import useState x
+ *    isLoading: false, x
+      email: "", x 
+      password: "", x
+      confirmPassword: "", x
+      confirmationCode: "", x
+      newUser: null, x
 
       bind forms to several useStates
  * 
- * import useHistory
+ * import useHistory x 
  * history.push("/onboarding/user");
  * 
  * 
  
  */
 
-export default class Signup extends Component {
-  constructor(props) {
-    super(props);
-
-    this.state = {
-      isLoading: false,
-      email: "",
-      password: "",
-      confirmPassword: "",
-      confirmationCode: "",
-      newUser: null,
-    };
-  }
+      
+const Signup = () => {
+  const [email, setEmail] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+  const [password, setPassword] = useState('');
+  const [confirmPassword, setConfirmPassword] = useState('');
+  const [confirmationCode, setConfirmationCode] = useState('');
+  const [newUser, setNewUser] = useState(null);
+
 
-  validateForm() {
+  // for redirect to new route
+  const history = useHistory()
+
+  const validateForm = () => {
     return (
-      this.state.email.length > 0 &&
-      this.state.password.length > 0 &&
-      this.state.password === this.state.confirmPassword
+      email.length > 0 &&
+      password.length > 0 &&
+      password === confirmPassword
     );
   }
 
-  validateConfirmationForm() {
-    return this.state.confirmationCode.length > 0;
+  const validateConfirmationForm = () => {
+    return confirmationCode.length > 0;
   }
 
-  handleChange = (event) => {
-    this.setState({
-      [event.target.id]: event.target.value,
-    });
-  };
-
-  handleSubmit = async (event) => {
+  const handleConfirmationSubmit = async (event) => {
     event.preventDefault();
 
-    this.setState({ isLoading: true });
+    setIsLoading(true)
 
     try {
-      const newUser = await Auth.signUp({
-        username: this.state.email,
-        password: this.state.password,
-      });
-      this.setState({
-        newUser,
-      });
+      await Auth.confirmSignUp(email, confirmationCode);
+      await Auth.signIn(email, password);
```

Comment:
After Auth.signIn, can you run the `successToast` function that is also exported from the libs toast file? We want to give our users some sort of visual proof that their request has either succeeded, or failed.

Reply:
Ok, what param do you want me to feed it with ?